### PR TITLE
[SDPA] Add expanded autograd testing for fused kernels and disable head_dim128 sm86 mem-efficient

### PR DESCRIPTION
**Pull request by drisspg**
# Summary
- Adds a large parameter sweep for testing the various configs a user can call sdpa with and compares the deviation of the fused kernels vs the eager math fallback to test for correctness.
- Sm86 + head_dim==128 is throwing an IMA  for memory efficient attention. We add a filter for use_mem_efficient_attention().  This has since been fixed in the upstream Xformers version but will likely not make it for branch cut.

### CODE REVIEW

**Review by cpuhrsch — APPROVED** [review]

File: test/test_transformers.py [hunk 1]
```diff
@@ -1581,6 +1593,88 @@ def func():
...
+
+        upstream_grad = torch.rand_like(out, requires_grad=False)
+
+        out.backward(upstream_grad)
+        out_ref.backward(upstream_grad.to(out_ref.dtype))
+        out_lp_ref.backward(upstream_grad.to(out_lp_ref.dtype))
+
+        # [Note] Fused Tolerances
+        # Establish the numerical error between the "true" high precision math output
+        # and the low precision math reference. We use this reference for the rtol
```
Comment (drisspg) [inline test/test_transformers.py:1651]: Rtol and atol should be swapped

**Pull request closed without merging**